plugins/blocklayer/plugin.go

```diff
@@ -114,7 +114,8 @@ func configure(plugin *node.Plugin) {
 
 	deps.Tangle.Booker.Events.BlockBooked.Attach(event.NewClosure(func(event *tangleold.BlockBookedEvent) {
 		deps.Tangle.Storage.Block(event.BlockID).Consume(func(block *tangleold.Block) {
-			deps.Tangle.WeightProvider.Update(block.IssuingTime(), identity.NewID(block.IssuerPublicKey()))
+			ei := epoch.IndexFromTime(block.IssuingTime())
+			deps.Tangle.WeightProvider.Update(ei, identity.NewID(block.IssuerPublicKey()))
```

Comment:
We update the weight provider in two places: on `BlockBooked` here, and on `BlockAccepted` in `acceptance_gadget.go:34`. I believe it should only be updated on Booking... let me check to make sure, but leaving a comment anyway to not forget.

Reply:
We had the same in the old code, so I guess this is fine!